pkg/tools/etcd_helper.go

```diff
@@ -165,12 +165,14 @@ func (h *EtcdHelper) ExtractObjToList(key string, listObj runtime.Object) error
 	}
 
 	nodes := make([]*etcd.Node, 0)
-	nodes = append(nodes, response.Node)
+	if !IsEtcdNotFound(err) {
```

Comment:
@smarterclayton I found that if there is an etcd not found error here (type 100) then response will be null so response.Node or response.EtcdIndex (below) will cause this goroutine to panic.